protoc-gen-gogo/generator/generator.go

```diff
@@ -2178,6 +2194,31 @@ func (g *Generator) generateMessage(message *Descriptor) {
 				log.Printf("don't know how to generate constant for %s", fieldname)
 				continue
 			}
+
+			// hunt down the actual enum corresponding to the default
+			var enumValue *descriptor.EnumValueDescriptorProto
+			for _, ev := range enum.Value {
+				// TODO: This could probably be preindexed but the performance
+				// benefit might not outweigh keeping tracking of the uniqueness
+				// across package boundaries.
+				if def == ev.GetName() {
+					enumValue = ev
+				}
+			}
+
+			if enumValue != nil {
+				if gogoproto.IsEnumValueCustomName(enumValue) {
+					def = gogoproto.GetEnumValueCustomName(enumValue)
+				}
+			} else {
+				// This condition should be rare or impossible. Could be caused
+				// by a compiler bug that didn't ensure that default enum value
+				// is define for the type.
+				g.Fail(fmt.Sprintf("could not resolve default enum value for %v.%v",
+					g.DefaultPackageName(obj), def))
+
+			}
+
 			if gogoproto.EnabledGoEnumPrefix(enum.file, enum.EnumDescriptorProto) {
```

Comment:
Maybe this if should be inside the enum prefix method?
I know its my bad code, but could you try to move it and see if its better?